models/migrations/v226.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"code.gitea.io/gitea/modules/timeutil"
+	"xorm.io/xorm"
+)
+
+// addConfidentialColumnToOAuth2ApplicationTable: add Confidential column, setting existing rows to true
```

Comment:
> setting existing rows to true

How can I achieve this? How can I test this?

Reply:
To do this you'd have to add the column to the table - without a default value set.
Then update everything in the table where `confidential IS NULL` to set it to `true` then update the table to set the default to false and not null. 

It would be easier to make it so that the default is the current behaviour.

Reply:
Thanks, that worked.